table/block_based/block_based_table_reader_test.cc

```diff
@@ -990,6 +993,118 @@ TEST_P(BlockBasedTableReaderTestVerifyChecksum, ChecksumMismatch) {
   ASSERT_EQ(s.code(), Status::kCorruption);
 }
 
+TEST_P(BlockBasedTableReaderTest, MultiScanPrepare) {
```

Comment:
Would be good to also add some tests to sanity check cases where the Seeks are not in the order specified in Prepare()